SubTemplates/IoT/Lambdas/provision_device/client/main.py

```diff
@@ -1,49 +1,84 @@
 # ------------------------------------------------------------------------------
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: MIT-0
+# SPDX-License-Identifier: MIT-0 
```

Comment:
I tried running with serial number. The client tried to create a thing with name 8675309 even though the serial number in machine_config was 12345ABCD. 

Connecting with official cert timed out

Reply:
I originally took out the file based configuration for serial numbers but after some thought, added it back in. So your expected behavior should work in the latest checkin on the feature branch.